include/output_bout_types.hxx

```diff
@@ -0,0 +1,47 @@
+///
+/// Implement {fmt} formatters for BOUT++ types
+///
+
+#ifndef OUTPUT_BOUT_TYPES_H
+#define OUTPUT_BOUT_TYPES_H
+
+#include <fmt/format.h>
+
+#include "output.hxx"
+#include "bout/region.hxx"
+
+template <IND_TYPE N>
+struct fmt::formatter<SpecificInd<N>> {
+  // Presentation format: 'c' - components, 'i' - index.
+  char presentation = 'c';
+
+  // Parses format specifications of the form ['c' | 'i'].
+  constexpr auto parse(format_parse_context& ctx) {
+    auto it = ctx.begin(), end = ctx.end();
+    if (it != end && (*it == 'c' || *it == 'i')) presentation = *it++;
+
+    // Check if reached the end of the range:
+    if (it != end && *it != '}')
+      throw format_error("invalid format");
+
+    // Return an iterator past the end of the parsed range:
+    return it;
+  }
+
+  // Formats the point p using the parsed format specification (presentation)
+  // stored in this formatter.
+  template <typename FormatContext>
+  auto format(const Ind3D& ind, FormatContext& ctx) {
```

Comment:
Worth adding `Ind2D` as well?

Reply:
Instead of overloading, we cpi;d take a `SpecificInd<N>` here, and then check `N == 3` to find out whether we use `ind.z()` or not.

Reply:
This was pretty easy to do with a switch-case to work for `Ind2D` and `IndPerp` as well.

@bendudson Are you ok if I push straight to this branch? I might also move the header to `include/bout` if that's ok.

Reply:
Great, thanks @ZedThree . Yes, happy for you to push straight to this. Agree probably best to put in `include/bout`.